### PR TITLE
[Bug] Profile Picture Not Getting Saved and Login Again & again fixed.

### LINKED ISSUE #96: [Bug] Profile Picture Not Getting Saved and Login Again & again
## Description
- [ ] The profile picture isn't getting saved in DB if i select an image during the brand registration
- [ ] After logging in, if i go to back/home page, i again need to signIn, if my token isn't expired, i should be directed to the loggedIn page ideally.
- [ ] When user revisits influencer registration form to make changes then social profile , phot0s etc fields data go reset and he has to reverify them all to go forward ..ideally after completion of process or if user doesnot want to register and w goes back then only fields data should reset ...until then all data should not be lost.
- [ ] Similarly for brands also once user goes back of regsitration form then data should be reset or user completes the process then data should reset

### PR DESCRIPTION
**Pull request by Hemu21**
## Description:

 - [x] The profile picture isn't getting saved in DB if i select an image during the brand registration
 - [x] After logging in, if i go to back/home page, i again need to signIn, if my token isn't expired, i should be directed to the loggedIn page ideally.
 - [x] When user revisits influencer registration form to make changes then social profile , phot0s etc fields data go reset and he has to reverify them all to go forward ..ideally after completion of process or if user doesnot want to register and w goes back then only fields data should reset ...until then all data should not be lost.
 - [x] Similarly for brands also once user goes back of regsitration form then data should be reset or user completes the process then data should reset

## Issue 

[Issue Link](https://github.com/Kajol-Kumari/InfluMart/issues/96) resolve #96 

## Screenshot or Video

https://github.com/Kajol-Kumari/InfluMart/assets/106808387/6a4efac9-9d49-4496-b2d0-9a0ea0c9a5ea

### CODE REVIEW

**Comment by Hemu21** [comment]
@Kajol-Kumari can you please check

File: frontend/screens/BrandProfile.js [hunk 1]
```diff
@@ -76,8 +76,7 @@ const BrandProfile = ({ navigation }) => {
     <ScrollView style={styles.container}>
       <View style={styles.container}>
         <View style={styles.header}>
-          <TouchableOpacity
-            onPress={() => navigation.navigate("LoginPageBrands")}
+          <TouchableOpacity
```
Comment (Kajol-Kumari) [inline frontend/screens/BrandProfile.js:79]: I am not able to go back after logging in, is this change intentional?

File: frontend/App.js [hunk 2]
```diff
@@ -56,6 +56,32 @@ const App = () => {
     "PlusJakartaSans-Bold": require("./assets/fonts/PlusJakartaSans-Bold.ttf"),
     "PlusJakartaSans-ExtraBold": require("./assets/fonts/PlusJakartaSans-ExtraBold.ttf"),
   });
+  const [initialRoute, setInitialRoute] = React.useState('Login');
+
+  React.useEffect(() => {
+    checkTokenValidity();
+  }, []);
+
+  const checkTokenValidity = async () => {
```
Comment (Kajol-Kumari) [inline frontend/App.js:65]: I am still seeing login option if i am refreshing the page

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
- [ ] The back button after logging in as brand not working, please check it.
- [ ] The brand signUp is continuously failing, please check the same. I tried without image as well and it still failed

Screenshot 2024-06-26 at 2 46 46 PM

- [ ] If i am going back from `preview` page, i am seeing this which is looking bad, it should just show and alert and take to signUp page

Screenshot 2024-06-26 at 2 48 36 PM

File: frontend/screens/BrandProfile.js (hunk 1 above)
Comment (Hemu21) [reply frontend/screens/BrandProfile.js:79]: you said after login you not want to see login or anything related to signup. so i changed it.

File: frontend/App.js (hunk 2 above)
Comment (Hemu21) [reply frontend/App.js:65]: ok wait i will check and update.

**Comment by Hemu21** [comment]
@Kajol-Kumari  updated can you please check.

**Comment by Kajol-Kumari** [comment]
Screenshot 2024-06-26 at 7 14 00 PM
@Hemu21 the signup is still giving internal server error.

**Comment by Hemu21** [comment]
@Kajol-Kumari  it is problem your deployed backend. For me it is working in localhost

**Comment by Kajol-Kumari** [comment]
Screenshot 2024-06-26 at 8 39 16 PM

I can see this in logs

**Comment by Hemu21** [comment]
@Kajol-Kumari  it indicates no folder named uploads/brandProfiles/ is existed in the repo. you have to create `uploads/brandProfiles/` and `uploads/influencerProfiles/` in the backend folder int the repo. while pushing the code those folder won't get included why because those folders doesn't have any file.

**Comment by Kajol-Kumari** [comment]
@Hemu21 yeah, i figured after seeing the error but the logic should have been for creation of the folder if it doesn't exists, anyway for now i have manually created the folder

**Review by Kajol-Kumari — APPROVED** [review]
LGTM 👍 
@RohitGupta1237 feel free to merge it after verifying the ui once

**Pull request merged**